packages/react-router-dom/index.tsx

```diff
@@ -1291,8 +1334,8 @@ function useScrollRestoration({
       }
 
       // otherwise go to the top on new locations
-      window.scrollTo(0, 0);
-    }, [location, restoreScrollPosition, preventScrollReset]);
+      scrollY(scrollContainer, 0);
+    }, [location, restoreScrollPosition, preventScrollReset, scrollContainer]);
```

Comment:
```suggestion
    }, [location, router.state, scrollContainer]);
```

One thing I was having trouble with after finally getting this all set up correctly was that scroll position wasn't getting restored on page reload.  
I was able to get it to work by doing this ☝🏻 , then referencing `router.state.restoreScrollPosition` and `router.state.preventScrollReset` in this `useLayoutEffect` handler instead of `restoreScrollPosition` and `preventScrollReset` directly.

Reply:
I think this is a function of the bug you pointed out above, since `savedScrollPositions` was always truthy, we were never re-filling it from from `sessionStorage` on reloads.

Reply:
@brophdawg11, confirmed.  All good now